integrations/launchdarkly/client.py

```diff
@@ -0,0 +1,169 @@
+from port_ocean.utils import http_async_client
+import httpx
+from typing import Any, AsyncGenerator, Optional
+from loguru import logger
+from enum import StrEnum
+
+PAGE_SIZE = 20
+
+
+class ResourceKindsWithSpecialHandling(StrEnum):
+    FEATURE_FLAG = "flag"
+    ENVIRONMENT = "environment"
+
+
+class ObjectKind(StrEnum):
+    PROJECT = "project"
+    AUDITLOG = "auditlog"
+
+
+class LaunchDarklyClient:
+    def __init__(
+        self, api_token: str, launchdarkly_url: str = "https://app.launchdarkly.com"
+    ):
+        self.api_url = f"{launchdarkly_url}/api/v2"
+        self.api_token = api_token
+        self.http_client = http_async_client
+        self.http_client.headers.update(self.api_auth_header)
+
+    @property
+    def api_auth_header(self) -> dict[str, Any]:
+        return {
+            "Authorization": f"{self.api_token}",
+            "Content-Type": "application/json",
+        }
+
+    async def get_paginated_resource(
+        self, kind: str, resource_path: str | None = None

```

Comment:
can you make the kind type be of either ResourceKindsWithSpecialHandling or ObjectKind instead of a str?

Reply:
Also, since the environments and feature flags depend on the projects, we need to introduce caching (for the projects) to make the application run fast.